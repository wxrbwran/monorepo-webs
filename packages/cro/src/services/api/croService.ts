import http from '@/services/http';
import { CommonData } from 'typings/global';

export default {
  // 我的团队 查询待添加的医生列表, 多选医生作为好友
  fetchBaseRoles(data: CommonData): Promise<any> {
    return http.get('base/roles', { data });
  },

  // 我的团队 查询好友列表
  fetchDoctorFriends(data: CommonData): Promise<any> {
    return http.get('doctor/friends', { data });
  },

  // 我的团队 查询好友列表
  putTeamMembers(data: CommonData): Promise<any> {
    return http.put('doctor/team/members', { data });
  },

  // 我的团队 查询好友列表
  getTeams(data: CommonData): Promise<any> {
    return http.get(`doctor/teams?data=${JSON.stringify(data)}`);
  },




};