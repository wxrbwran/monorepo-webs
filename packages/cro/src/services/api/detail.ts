import http from '@/services/http';

export default {
  fetchMedicines(data: any): Promise<any> {
    return http.get(`medicine/name?data=${JSON.stringify(data)}`);
  },
  fetchDiagnosis(data: any): Promise<any> {
    return http.get(`disease?data=${JSON.stringify(data)}`);
  },
  // fetchUploadInfo(file: object): Promise<any> {
  //     return http.get(`clinical/form/address?fileName=${file.name}`);
  // },
  fetchUploadInfo(data?: any): Promise<any> {
    return http.get(`cro/file/address?projectSid=${data.projectSid}&fileName=${data.fileName}`);
  },
  fetchTreatment(data: any): Promise<any> {
    return http.get(`treatment?data=${JSON.stringify(data)}`);
  },
  // 获取项目详情
  getCroProjectInfo(value: string): Promise<any> {
    return http.get(`research/project/detail/${value}`);
  },
  // 添加文件
  addProjectFile(data?: any): Promise<any> {
    return http.post('research/project/file', { data });
  },
  // 文件列表
  getProjectFileList(data?: any): Promise<any> {
    return http.get(`research/project/file?data=${JSON.stringify(data)}`);
  },
  // 删除文件
  deleteProfileFile(fileId: string): Promise<any> {
    return http.delete(`research/project/file/${fileId}`);
  },
  //更新项目信息
  updateCroProject(data?: any): Promise<any> {
    return http.patch('research/project', { data });
  },
  //添加加入、排除标准
  addStandard(data?: any): Promise<any> {
    return http.post('cro/project/standard', { data });
  },
  //查看加入、排除标准
  getStandard(value: string): Promise<any> {
    return http.get(`cro/project/standard/${value}`);
  },
  //获取客观量表（检查提醒）
  getObjectiveList(data?: any): Promise<any> {
    return http.get(`research/objective/examination/scale/${data}`);
  },
  //预览文件
  getFileInfo(fileId: any): Promise<any> {
    return http.get(`research/project/file/info/${fileId}`);
  },
};
