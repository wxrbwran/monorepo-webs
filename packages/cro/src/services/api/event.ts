import http from '@/services/http';

export default {
  // 添加项目终点事件
  addEndEvent(data?: any): Promise<any> {
    return http.post('research/project/end_event',{data});
  },
  // 查看项目终点事件
  fetchEndEvent(projectSid: any): Promise < any > {
    return http.get(`research/end_event/detail/${projectSid}`);
  },
  // 修改项目终点事件
  updateEndEvent(data?: any): Promise < any > {
    return http.patch('research/project/end_event', { data });
  },
  // 查询项目中所有研究者
  fetchResearchers(projectNsId: any): Promise < any > {
    return http.get(`research/project/researchers/${projectNsId}`);
  },
  // 查询项目终点事件统计信息
  fetchEventCountInfo(data?: any): Promise < any > {
    return http.get(`research/project/event_statistics?data=${JSON.stringify(data)}`);
  },
}
