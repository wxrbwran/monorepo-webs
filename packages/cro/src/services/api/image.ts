import http from '@/services/http';

export default {
  // 获取化验单检查单统计数据接口new
  // fetchImageCountNew(data: any): Promise<any> {
  //   return http.get('image/count/last', { data });
  // },
  // 获取具体类型化验单检查单图片
  // fetchImageDetail(data: any): Promise<any> {
  //   return http.get('image/detail', { data });
  // },
  // // 获取具体类型化验单检查单图片V1版接口
  // fetchImageDetailV1(data: any): Promise<any> {

  //   return http.get('image/detail/v1', { data });
  // },
  // // 修改图片旋转角度
  // patchImageDegree(data: any): Promise<any> {
  //   return http.patch('image/detail', { data });
  // },
  // // 获取自定义指标历史中所有的图片分类
  // fetchImageTypeTabs(data: any): Promise<any> {
  //   return http.get('image/type/last', { data });
  // },
  // // 获取某一项图片分类的所有指标详情
  // fetchImageIndexHistory(data: any): Promise<any> {
  //   return http.get('image/index/history/last', { data });
  // },
  // 获取图片结构化数据指标
  // fetchImageIndex(data: any): Promise<any> {

  //   return http.get('nurse/image/index', { data });
  // },
  // // 保存图片结构化数据指标
  // putImageIndex(data: any): Promise<any> {
  //   return http.put('nurse/image/index', { data });
  // },

  // // 获取图片结构化接口
  // fetchImageIndexCustom(data: any): Promise<any> {
  //   return http.get('nurse/image/index/custom', { data });
  // },
  // // 修改图片结构化数据指标
  // putImageIndexCustom(data: any): Promise<any> {
  //   return http.put('nurse/image/index/custom', { data });
  // },

  // // 获取所有子分类，即检查部位
  // fetchImageSubtypes(): Promise<any> {
  //   return http.get('image/subtypes');
  // },
  // // 搜索图片大分类或者指标
  // fetchImageIndexInfo(data: any): Promise<any> {
  //   return http.get('image/index/info', { data });
  // },
  // // 结构化
  // // 获取图片信息及结构化数据
  // putImage(data: any): Promise<any> {
  //   return http.put('image', { data });
  // },
  fetchImageIndexes(data: any): Promise<any> {
    return http.get(`image/indexes?data=${JSON.stringify(data)}`);
    // return http.get('image/indexes', { data });
  },
  // // 保存图片信息及结构化数据
  // putImageImageIndexes(data: any): Promise<any> {
  //   return http.put('image/indexes', { data });
  // },
  // 获取检查单、其它单据
  fetchImageJcdAndOther(data: any): Promise<any> {
    // return http.get('image/jcd', { data });
    return http.get(`image/jcd?data=${JSON.stringify(data)}`);
  },
  // // 新增检查单、其它单据
  // putImageJcdAndOther(data: any): Promise<any> {
  //   return http.put('image/jcd', { data });
  // },
  // // 获取检查单问题模板
  // fetchImageTopicTemplate(data: any): Promise<any> {
  //   return http.get('image/template', { data });
  // },
  // // 新增检查单问题模板   11.18新增系统/医生检查单
  // putImageTopicTemplate(data: any): Promise<any> {
  //   return http.put('image/template', { data });
  // },
  // // 新增/修改/删除检查单问题
  // postImageTemplate(data: any): Promise<any> {
  //   return http.post('image/template', { data });
  // },
  // // 修改检查单模板名称
  // patchImageTemplateName(data: any): Promise<any> {
  //   return http.patch('image/template', { data });
  // },
  // // 获取检查单检查部位或检查方法
  // fetchImageTemplatePart(): Promise<any> {
  //   return http.get('image/template/part');
  // },
  // // 根据检查部位或检查方法获取检查名称
  // fetchImageTemplateName(data: any): Promise<any> {
  //   return http.get('image/template/name', { data });
  // },
  // 结构化
};
