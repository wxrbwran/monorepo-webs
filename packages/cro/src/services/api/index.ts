export { default as auth } from './auth';
export { default as detail } from './detail';
export { default as user } from './user';
export { default as subjective } from './subjective';
export { default as project } from './project';
export { default as patientManage } from './patientManage';
export { default as report } from './report';
export { default as query } from './query';
export { default as template } from './template';
export { default as research } from './research';
export { default as base } from './base';
export { default as event } from './event';
export { default as service } from './croService';
export { default as image } from './image';