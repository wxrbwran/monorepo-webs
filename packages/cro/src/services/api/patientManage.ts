
import http from '@/services/http';

export default {
  // 获取全部患者
  getPatientList(data: any): Promise<any> {
    return http.get(`research/teams/patient?data=${JSON.stringify(data)}`);
  },
  // 获取受试者列表
  getTestPatients(data: any): Promise<any> {
    return http.get(`research/teams/project/patient?data=${JSON.stringify(data)}`);
  },
  // 全部患者列表医生手动打钩 勾选患者
  postCheckPatient(data: any): Promise<any> {
    return http.post('research/patient/checked', { data });
  },

  // 添加或修改 受试者在项目的给药时间或停止此项目用药时间
  patchPatientMedTimeStatus(data: any): Promise<any> {
    return http.patch('research/project/patient/med_time', { data });
  },
  // research/project/patient/med_time


  //全部患者列表医生去除打钩
  // patchDelCheckPatient(data: any): Promise<any> {
  //   return http.patch('cro/project/check/patient/type', { data });
  // },
  // 创建试验小组
  postAddGroup(data: any): Promise<any> {
    return http.post('research/project/group', { data });
  },
  // 获取小组列表
  getGroupList(projectNsId: any): Promise<any> {
    return http.get(`research/group/${projectNsId}`);
  },
  // 获取入组排除标准列表
  getStandard(projectSid: any): Promise<any> {
    return http.get(`cro/project/standard/${projectSid}`);
  },
  // 查看项目文件列表
  getProjectFile(projectSid: any, data: any): Promise<any> {
    return http.get(`cro/project/file/${projectSid}`, { data });
  },
  // 发送试验项目书、知情同意书 发送项目邀请
  postSendFile(data: any): Promise<any> {
    return http.post('research/project/invite', { data });
  },
  // 私下已签署，直接添加患者为受试者
  postJoinProject(data: any): Promise<any> {
    return http.post('research/project/join/by_doctor', { data });
  },
  // 停止参与试验 更改受试者状态
  patchPatientStatus(data: any): Promise<any> {
    return http.patch('research/project/patient', { data });
  },
  // 向小组添加患者 加入实验分组
  postGroupPatient(data: any): Promise<any> {
    return http.post('research/group/patient', { data });
  },
  // 查询小组内患者
  getGroupPatient(data: any): Promise<any> {
    return http.get(`research/group/patient?data=${JSON.stringify(data)}`);
  },
  //修改试验小组名
  modifyGroup(data: any): Promise<any> {
    return http.patch('research/project/group', { data });
  },
};
