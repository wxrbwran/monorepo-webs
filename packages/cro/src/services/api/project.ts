import http from '@/services/http';

export default {
  // 创建项目
  postCroProject(data: any): Promise<any> {
    return http.post('research/project', { data });
  },
  // 获取项目列表
  getProjectList(data: any): Promise<any> {
    // return http.get(`research/all/projects?data=${JSON.stringify(data)}`);
    // return http.get(`research/all/projects?doctorId=${1381}`);
    return http.get(`research/all/projects`);
  },
  // 获取患者统计图表
  getPatientStatistic(): Promise<any> {
    return http.get(`research/patient/statistics`);
  },
  // 获取项目统计图表
  getPStatistic(): Promise<any> {
    return http.get(`research/project/statistics`);
  },
  //删除项目
  delProject(data: any): Promise<any> {
    return http.delete(`research/project/${data}`);
  },
};
