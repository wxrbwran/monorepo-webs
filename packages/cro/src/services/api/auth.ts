import { auth, ajax } from '@/services/auth';
import qs from 'qs';

export default {
  // 登录
  token(formData: CommonData): Promise<any> {
    return auth.post('token', {
      data: qs.stringify(formData)
    });
  },
};
