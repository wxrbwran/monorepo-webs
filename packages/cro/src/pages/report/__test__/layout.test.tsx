import * as React from 'react'
import { render, RenderResult, act, waitFor } from '@testing-library/react';
import EmptyReport from '../index';
import Layout from '../_layout';
import * as api from '@/services/api';
import { BrowserRouter as Router } from 'react-router-dom';
jest.mock("@/services/api", () => {
  return {
    report: {
      getProjectList: jest.fn(() => {
        return Promise.resolve({ apiInfos: [], status: 'success' });
      })
    }
  };
});
let wrapper: RenderResult;
describe('报告的layout', () => {
  const emptyProps = {
    location: {
      pathname: 'report',
      query: {
        reportId: '',
      },
    },
  }
  const detailProps = {
    location: {
      pathname: 'report/detail',
      query: {
        reportId: '5Z7g7r',
      },
    },
  }
  it('匹配快照', () => {
    expect(wrapper).toMatchSnapshot();

  })
  it('不存在报告列表时，左侧不显示报告列表', async() => {
    wrapper = render(<Router>
      <Layout {...emptyProps}>
        <EmptyReport />
      </Layout>
    </Router >)
    await waitFor(() => {
      expect(api.report.getReportList).toBeCalled();
    })
    expect(wrapper.container.querySelector('.table-list')).toBeNull();
  })
})
