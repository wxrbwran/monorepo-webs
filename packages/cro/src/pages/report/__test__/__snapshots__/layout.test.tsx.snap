// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`报告的layout 匹配快照 1`] = `undefined`;
